RecoEgamma/ElectronIdentification/plugins/cuts/GsfEleMVAExpoScalingCut.cc

```diff
@@ -0,0 +1,128 @@
+#include "PhysicsTools/SelectorUtils/interface/CutApplicatorWithEventContentBase.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+
+#include <math.h>
+
+class GsfEleMVAExpoScalingCut : public CutApplicatorWithEventContentBase {
+public:
+  GsfEleMVAExpoScalingCut(const edm::ParameterSet& c);
+
+  result_type operator()(const reco::GsfElectronPtr&) const final;
+
+  void setConsumes(edm::ConsumesCollector&) final;
+  void getEventContent(const edm::EventBase&) final;
+
+  double value(const reco::CandidatePtr& cand) const final;
+
+  CandidateType candidateType() const final { 
+    return ELECTRON; 
+  }
+
+private:
+
+  // Cut values
+  const std::vector<double> mvaCutValues;
+
+  // Pre-computed MVA value map
+  edm::Handle<edm::ValueMap<float> > mvaValueMap;
+  edm::Handle<edm::ValueMap<int> > mvaCategoriesMap;
+
+};
+
+DEFINE_EDM_PLUGIN(CutApplicatorFactory,
+                  GsfEleMVAExpoScalingCut,
+                  "GsfEleMVAExpoScalingCut");
+
+GsfEleMVAExpoScalingCut::GsfEleMVAExpoScalingCut(const edm::ParameterSet& c) :
+  CutApplicatorWithEventContentBase(c),
+  mvaCutValues(c.getParameter<std::vector<double> >("mvaCuts"))
+{
+  edm::InputTag mvaValTag = c.getParameter<edm::InputTag>("mvaValueMapName");
+  contentTags_.emplace("mvaVal",mvaValTag);
+  
+  edm::InputTag mvaCatTag = c.getParameter<edm::InputTag>("mvaCategoriesMapName");
+  contentTags_.emplace("mvaCat",mvaCatTag);
+  
+}
+
+void GsfEleMVAExpoScalingCut::setConsumes(edm::ConsumesCollector& cc) {
+
+  auto mvaVal = 
+    cc.consumes<edm::ValueMap<float> >(contentTags_["mvaVal"]);
+  contentTokens_.emplace("mvaVal",mvaVal);
+
+  auto mvaCat = 
+    cc.consumes<edm::ValueMap<int> >(contentTags_["mvaCat"]);
+  contentTokens_.emplace("mvaCat",mvaCat);
+}
+
+void GsfEleMVAExpoScalingCut::getEventContent(const edm::EventBase& ev) {  
+
+  ev.getByLabel(contentTags_["mvaVal"],mvaValueMap);
+  ev.getByLabel(contentTags_["mvaCat"],mvaCategoriesMap);
+}
```

Comment:
why not getByToken rather than getByLabel?

Reply:
I tried, but ev is a const edm::EventBase& and not Event. This guy has not getByToken method, as I am told by the compiler. Is that correct or am I just doing something wrong? Is there a way I can work around this to still use getByToken?

Reply:
Also, I found no example in CMSSW where a cut was derived from PhysicsTools/SelectorUtils/CutApplicatorWithEventContentBase, so I could not get inspired by that...

Reply:
hmm, that is odd, okay fine to leave as is for now